### PR TITLE
Add parenthesis for binary operators

### LINKED ISSUE #872: Potentially include parentheses on expressions mixing math and binary operators
Another case like https://github.com/prettier/prettier/issues/187 that I ran into in https://github.com/babel/babel/pull/5412

```js
const result = (a + b) >>> 1;
```
becomes
```js
const result = a + b >>> 1;
```

and I'd be very surprised if anyone remembers the precedence for binary math operators.
 
Thoughts?

### PR DESCRIPTION
**Pull request by vjeux**
Fixes #872

### CODE REVIEW

**Comment by lydell** [comment]
What about mixing binary operators with other binary operators?

**Comment by vjeux** [comment]
I'm a bit worried that people that do write a lot of binary operators would be annoyed by this. But it's very unclear how precedence work for them. I just looked at a piece of code I wrote many years ago and I over-parenthesized!

https://github.com/jDataView/jDataView/blob/master/src/jdataview.js#L562-L564

I'll update this PR.

**Comment by vjeux** [comment]
Done

File: tests/binary_math/jsfmt.spec.js [hunk 1]
```diff
@@ -0,0 +1 @@
+run_spec(__dirname);
```
Comment (existentialism) [inline tests/binary_math/jsfmt.spec.js]: Should we add babylon?

Comment (vjeux) [inline tests/binary_math/jsfmt.spec.js]: Sure. I don't expect babylon to be anything different here tbh

**Comment by jlongster** [comment]
👍 Sounds good to me!

**Pull request merged**